mobify/commercecloud-ocapi-client
# Catch JSON.parse errors when handling 'callApi' error response text

# Description

On Addition Elle, we saw the OCAPI backend return 404's/500's with (presumably) HTML response bodies. This caused the SSR server rendering to hang, so it responded with 502's after the 20 second timeout.

Here, we now catch JSON parse errors so that the SSR server will still respond with a rendered page.

## Type of change


## Changes

- Wrapped `JSON.parse()` call in a try/catch block

# How to test this PR?

- Checkout the `additionelle` repo
- In this repo, run `npm link`
- In `additionelle/web`, run `npm link commercecloud-ocapi-client`
- In `additionelle/web/app/store/index.js`, change `getConnectorBasePath()` to return `"/page-not-found"`
   - This will cause the OCAPI requests to point at `/page-not-found`, which returns a 404 and includes an HTML response body
- In `additionelle/web`, run `npm run local`
- Navigate to https://localhost:3443/en/apparel-pants-flared
- **Verify** the SSR server renders the page (doesn't hang)

# Checklist:

- [x] My code follows the style guidelines of this project (`npm run lint`)
- [x] New and existing unit tests pass locally with my changes (`npm test`)
